components/empty-state/src/empty-state.ts

```diff
@@ -197,48 +205,60 @@ export class DtEmptyState
   }
 
   ngAfterViewInit(): void {
-    this._viewportResizer
-      .change()
-      .pipe(startWith(null), takeUntil(this._destroy$))
-      .subscribe(() => {
-        this._updateLayout();
+    // Check if the broser supports the resizeObserver.
+    if (this._platform.isBrowser && 'ResizeObserver' in window) {
+      this._containerSizeObserver = new window.ResizeObserver(entries => {
+        // We need to wrap the call to the layout update into an additional
+        // requestAnimationFrame, because the resize observer would trow a
+        // javascript error when it is not able to process all entries within
+        // a single animation frame.
+        // From the specification:
+        // > This error means that ResizeObserver was not able to deliver all
+        // > observations within a single animation frame. It is benign (your
+        // > site will not break). - Aleksandar Totic
+        // https://stackoverflow.com/a/50387233
+        window.requestAnimationFrame(() => {
+          this._updateLayout(entries[0].contentRect.width);
```

Comment:
Is it possible that entries is empty or has no index 0?

Reply:
Not really, that would mean that the elementRef of the host element is empty. But I can add a check there, just to be sure.